partner_multi_relation/models/res_partner_relation_type_selection.py

```diff
@@ -120,28 +120,27 @@ def _auto_init(self):
         )
         return super()._auto_init()
 
-    def name_get(self):
+    @api.depends("is_inverse", "type_id.name_inverse", "type_id.display_name")
+    def _compute_display_name(self):
         """Get name or name_inverse from underlying model."""
-        return [
-            (
-                this.id,
+        for this in self:
+            this.display_name = (
                 this.is_inverse
                 and this.type_id.name_inverse
-                or this.type_id.display_name,
+                or this.type_id.display_name
             )
-            for this in self
-        ]
 
     @api.model
     def name_search(self, name="", args=None, operator="ilike", limit=100):
         """Search for name or inverse name in underlying model."""
         # pylint: disable=no-value-for-parameter
-        return self.search(
+        records = self.search(
             [
                 "|",
                 ("type_id.name", operator, name),
                 ("type_id.name_inverse", operator, name),
             ]
             + (args or []),
             limit=limit,
-        ).name_get()
+        )
+        return [(record.id, record.display_name) for record in records.sudo()]
```

Comment:
Is adding sudo necessary?

Reply:
Removed the sudo

Reply:
Squashed the requested commits and changed commit message